### PR TITLE
Add CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by Crissman**
## Motivation

Let people know how to contribute.

## Description of the changes

Add CONTRIBUTING.md file.

### CODE REVIEW

**Comment by Crissman** [comment]
Address #7

**Comment by Crissman** [comment]
@hvy Can you take a look at this?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+Code is formatted with [black](https://github.com/psf/black),
+Coding style is checked with [flake8](http://flake8.pycqa.org)
+and additional conventions are described in the [Wiki](https://github.com/optuna/optuna/wiki/Coding-Style-Conventions).
+
+If your environment misses some dependencies such as black and flake8.
+you will be asked to install them.
+
+## Continuous Integration and Local Verification
+
+Optuna repository uses GitHub Actions and CircleCI.
```
Comment (hvy) [inline CONTRIBUTING.md:34]:
```suggestion
This repository uses GitHub Actions.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+### Title
+
+The title of your pull request should
+
+- briefly describe and reflect the changes
+- wrap any code with backticks
+- not end with a period
+
+*The title will be directly visible in the release notes.*
```
Comment (hvy) [inline CONTRIBUTING.md:63]: I personally think we can omit this sentence. It's more confusing than informative. This repository itself doesn't provide "release notes".

**Review by hvy — CHANGES_REQUESTED** [review]
Left some nitpicks.

File: CONTRIBUTING.md (hunk 2 above)
Comment (hvy) [reply CONTRIBUTING.md:63]:
```suggestion
```

**Review by hvy — APPROVED** [review]
Thank you, LGTM!

**Pull request merged**